de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/filesystem/EclipseResourceImpl.java

```diff
@@ -8,12 +8,17 @@
 public class EclipseResourceImpl implements IResource {
 
     protected final org.eclipse.core.resources.IResource delegate;
+    protected IReferencePoint referencePoint;
 
     EclipseResourceImpl(org.eclipse.core.resources.IResource delegate) {
         if (delegate == null)
             throw new NullPointerException("delegate is null");
 
         this.delegate = delegate;
+
+        if (delegate.getProject() != null)
+            referencePoint = new EclipseReferencePointImpl(new EclipsePathImpl(
+                delegate.getProject().getLocation()));
```

Comment:
Using getLocation() you leave the VFS. My question is: Why do you want to not work with the VFS provided by the file system ? Do you need a unique reference ?

As you do not describe what an ReferencePoint is I have a hard time to understand the logic. So the reference point for each resource is its project ? I thought a reference point could be any container resource or just a file resource.

Reply:
I'm very sorry about the missing documentation of the IReferencePoint. After some feedbacks I rewrote the issue https://github.com/saros-project/saros/issues/177. There is explained, what a referencePoint is, for what we need this and especially why the interface is empty.

It is very confusing that the reference point actual gets the location of the project. The reason is, that Saros Core is project based (ex. activities). If Saros is not project dependent anymore, the UI will adjusted, so the user can choose any subtree of resources and the location of the subtree is the reference point.